robotoff/workers/tasks/import_image.py

```diff
@@ -215,6 +227,68 @@ def run_nutrition_table_object_detection(product_id: ProductIdentifier, image_ur
 }
 
 
+def run_upc_detection(product_id: ProductIdentifier, image_url: str):
+    """Detect the presence of a UPC in an image and determine if it dominates the image (i.e. if it is the primary content in the image)
+    :param product_id: identifier of the product
+    :param image_url: URL of the image to use
+    """
+    image = get_image_from_url(image_url, error_raise=False, session=http_session)
+
+    if image is None:
+        logger.info("Error while downloading image %s", image_url)
+        return
+
+    open_cv_image = numpy.array(image)
+
+    source_image = get_source_from_url(image_url)
+
+    pct_area, prediction_class, box = find_image_is_upc(
+        open_cv_image
+    )  # run upc detection
+
+    with db:
+        imagemodel = ImageModel.get_or_none(
+            source_image=source_image, server_type=product_id.server_type.name
+        )
+
+        if not imagemodel:
+            logger.info("Missing image in DB for image %s", source_image)
+            return
+
+        ImagePrediction.create(
+            image=imagemodel,
+            type="upc_image",
+            model_name="upc_opencv",
+            model_version="upc_opencv_1.0",
+            data={
+                "box": box.tolist(),
+                "area": pct_area,
+                "class": prediction_class.value,
+            },
+            max_confidence=None,
+        )
+
+        # no prediction neccessary if the image is not a UPC Image
+        if not prediction_class.value == "UPC_IMAGE":
+            return
+
+        prediction = Prediction(
+            type=PredictionType.is_upc_image,
+            value=None,
+            source_image=source_image,
+            server_type=product_id.server_type,
+            barcode=product_id.barcode,
+            # FIXME set to false
+            automatic_processing=True,
+            data={"pct_area": pct_area, "box": box.tolist()},
```

Comment:
Same comment about `box` -> `bounding_box`. Can you rename `pct_area` into something like `area`? It's better to avoid using abbreviations.

Reply:
Sure thing.